docs/UsersGuide/source/Glossary.rst

```diff
@@ -67,6 +67,9 @@ Glossary
    cycle-independent
       Describes a workflow task that only needs to be run once per experiment, regardless of the number of cycles in the experiment.
    
+   data assimilation
+      One of the major sources of error in weather and climate forecasts is uncertainty related to the initial conditions that are used to generate future predictions. Even the most precise instruments have a small range of unavoidable measurement error, which means that tiny measurement errors (e.g., related to atmospheric conditions and instrument location) can compound over time. These small differences result in very similar forecasts in the short term (i.e., minutes, hours), but they cause widely divergent forecasts in the long term. Data assimilation systems seek to mitigate this problem by combining the most timely observational data with other sources of data, such as historical data, to provide an analysis of possible atmospheric states and the probabilities of each. Errors in weather and climate forecasts can also arise because models are imperfect representations of reality. Data assimilation systems can use techniques including stochastic physics, which applies randomized perturbations to the physical tendencies or the physical parameters of a model, to compensate for model uncertainty. 
```

Comment:
@mkavulich Honestly, this is really helpful! I have not been super happy with this definition and was trying to revise for the Land DA docs. Do you think it would be helpful to add an introductory sentence along the lines of "Data assimilation is the process of combining different sources of data to account for forecast error and/or model uncertainty"? (Feel free to revise for accuracy.) I was thinking that the current definition jumps into why we need DA and what DA does without really saying what it IS first.

Reply:
I think a leading sentence along these lines would be great:
"Data assimilation is the process of combining observations, model data, and error statistics to achieve the best estimate of the state of a system." This explanation is more general and correct, since "data assimilation" can be done without running a forecast (for example, when performing a [reanalysis](https://en.wikipedia.org/wiki/Atmospheric_reanalysis)) and theoretically it has nothing to do with creating error/uncertainty estimates (though its use is often intertwined with error estimation, especially for ensemble DA methods). You could change "a system" to read "the atmosphere" if you want to keep it weather-specific, though in general it applies to any system (ocean or hydrology modeling, for example).

Reply:
Just added--thanks! :)